astarte-device-sdk-derive/src/case.rs

```diff
@@ -72,10 +76,12 @@ impl RenameRule {
     }
 
     /// Apply a renaming rule to a struct field, returning the version expected in the source.
-    pub fn apply_to_field(&self, field: &str) -> String {
+    pub fn apply_to_field<'a>(&self, field: &'a str) -> Cow<'a, str> {
         match *self {
-            RenameRule::None | RenameRule::LowerCase | RenameRule::SnakeCase => field.to_owned(),
-            RenameRule::UpperCase => field.to_ascii_uppercase(),
+            RenameRule::None | RenameRule::LowerCase | RenameRule::SnakeCase => {
```

Comment:
LowerCase and SnakeCase assume that the field is following Rust's field naming convention